plugins/modules/zos_replace.py

```diff
@@ -0,0 +1,716 @@
+        created. e.g. I(SOME.CREATION(-1)).
+    required: true
+    type: str
+    aliases: [ src, path, destfile ]
+  tmp_hlq:
+    description:
+      - Override the default High Level Qualifier (HLQ) for temporary and backup
+        data sets.
+      - The default HLQ is the Ansible user used to execute the module and if
+        that is not available, then the value of C(TMPHLQ) is used.
+    required: false
+    type: str
+  regexp:
+    description:
+      - The regular expression to look for in the contents of the file.
+    required: true
+    type: str
+  replace:
+    description:
+      - The string to replace I(regexp) matches with.
+      - If not set, matches are removed entirely.
+    required: false
+    type: str
+    default: ""
+"""
+
+EXAMPLES = r"""
+- name: Replace with blank space on a USS file any occurrences of the regex
+  zos_replace:
+    target: /tmp/src/somefile
+    regexp: 'profile\/'
+
+- name: Replace using after on USS file
+  zos_replace:
+    target: "/tmp/source"
+    regexp: '^MOUNTPOINT*'
+    after: export ZOAU_ROOT
+
+- name: Replace a specific line with special character on a dataset after a line
+  zos_replace:
+    target: SAMPLE.SOURCE
+    regexp: //*LIB  DD UNIT=SYS,SPACE=(TRK,(1,1)),VOL=SER=vvvvvv
+    replace:  //*LIB  DD UNIT=SYS,SPACE=(CYL,(1,1))
+    after: '^\$source base \([^\s]+\)'
+    disable_regex: regexp
+
+- name: Replace a specific line before a specific sentence with backup
+  zos_replace:
+    target: SAMPLE.SOURCE
+    backup: True
+    regexp: //SYSPRINT DD SYSOUT=*
+    before: SAMPLES OUTPUT SYSIN *=$DSN
+    disable_regex:
+        - regexp
+        - before
+
+- name: Replace some words between two lines with a backup with tmp_hlq
+  zos_replace:
+    target: SAMPLE.DATASET
+    tmp_hlq: ANSIBLE
+    backup: True
+    backup_name: BACKUP.DATASET
+    regexp: var
+    replace: vars
+    after: ^/tmp/source*
+    before: ^   if*
+
+- name: Replace lines on a GDS and generate a backup on the same GDG
+  zos_replace:
+    target: SOURCE.GDG(0)
+    regexp: ^(IEE132I|IEA989I|IEA888I|IEF196I|IEA000I)\s.*
+    after: ^IEE133I PENDING *
+    before: ^IEE252I DEVICE *
+    backup: True
+    backup_name: "SOURCE.GDG(+1)"
+
+- name: Delete some calls to SYSTEM on a member using a backref
+  zos_replace:
+    target: PDS.SOURCE(MEM)
+    regexp: '^(.*?SYSTEM.*?)SYSTEM(.*)'
+    replace: '\1\2'
+    after: IEE133I PENDING *
+    before: IEF456I JOB12345 *
+"""
+
+RETURN = r"""
+backup_name:
+    description: Name of the backup file or data set that was created.
+    returned: if backup=true
+    type: str
+    sample: /path/to/file.txt.2015-02-03@04:15
+changed:
+    description:
+        Indicates if the source was modified.
+    returned: always
+    type: bool
+    sample: 1
+found:
+    description: Number of matches found
+    returned: success
+    type: int
+    sample: 5
+msg:
+    description: Error messages from the module
```

Comment:
Is `msg` really used for **error** messages? I have not read down through the code but does not align to the Ansible doc for a module https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#msg